diff --git a/rsyslog.service.in b/rsyslog.service.in
index b761ae571..dfc125347 100644
--- a/rsyslog.service.in
+++ b/rsyslog.service.in
@@ -1,12 +1,14 @@
 [Unit]
 Description=System Logging Service
-Requires=syslog.socket
+;Requires=syslog.socket
+Wants=network.target network-online.target
+After=network.target network-online.target
 Documentation=man:rsyslogd(8)
 Documentation=https://www.rsyslog.com/doc/
 
 [Service]
 Type=notify
-ExecStart=@sbindir@/rsyslogd -n -iNONE
+ExecStart=@sbindir@/rsyslogd -n $SYSLOGD_OPTIONS
 StandardOutput=null
 Restart=on-failure
 
@@ -16,4 +18,4 @@ LimitNOFILE=16384
 
 [Install]
 WantedBy=multi-user.target
-Alias=syslog.service
+;Alias=syslog.service
